iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp

```diff
@@ -215,18 +222,56 @@ static bool haveEqualShapeDim(Value x, Value y, int i) {
          y.getType().cast<ShapedType>().getDimSize(i);
 }
 
+// Current instances of integer matmuls in linalg tend to use signless integers
```

Comment:
I think linalg is at the layer that should be using operations to convey sign information, not types. The linalg we consume has dialects-formerly-known-as-std (usually arith) in its regions, and those exclusively operate on signless.

Reply:
OK. I'll remove the comment. Should I leave the current code unchanged or change it to expect only signless integers?

Reply:
Done as discussed offline (signless only)